### ISSUE TITLE
Error om awsmobile run

### ISSUE THREAD

**Issue by halaesaili** [aws-samples/aws-mobile-appsync-chat-starter-angular#19]
I followed the tutorial exactly and I'm getting this error

> executing frontend start command ...
> frontend start command execution error:
> { Error: spawn ng ENOENT
>     at _errnoException (util.js:992:11)
>     at Process.ChildProcess._handle.onexit (internal/child_process.js:190:19)
>     at onErrorNT (internal/child_process.js:372:16)
>     at _combinedTickCallback (internal/process/next_tick.js:138:11)
>     at process._tickDomainCallback (internal/process/next_tick.js:218:9)
>   code: 'ENOENT',
>   errno: 'ENOENT',
>   syscall: 'spawn ng',
>   path: 'ng',
>   spawnargs: [ 'serve', '--host', '0.0.0.0', '--port', '4201' ] }

**Comment by dblemay** [comment]
I had the same problem.  Try using the Angular server instead of awsmobile.
`ng serve --open`

**Comment by VictorRed** [comment]
Did it works? I get exactly the same error

**Comment by dblemay** [comment]
Yes, in step 6 use `ng serve --open` instead of `awsmobile run` and it works for me.

**Comment by awsed** [comment]
Just committed a new package.json file with some new library versions. Please try again with the new versions, it should fix the issue

**Comment by awsed** [comment]
Closing due to inactivity. Issue should be fixed with latest commit.

**Comment by markl-vesper** [comment]
Hi @awsed 

I just pulled the latest master and had the same issue today when using awsmobile run cmd

using ng serve --open worked till it failed on another issue which I will try to fix first before opening a new issue. 

Just an FYI that it appears to still be an issue in the current master

**Comment by markl-vesper** [comment]
Just another quick comment for anyone playing around with this project. 

Should you want to involve other team members in the fun then use ng serve --open --host 0.0.0.0 --port 4200 to start the project and then they can browse to the webpage from their browser on their machine and you dont have to have multiple browsers open on your own machine to test. 

And you can actually test offline stuff way easily ;)

**Comment by naginagi** [comment]
@markl-vesper I assume you are using Windows, as this is a specific problem with NodeJS's spawn() on Windows, which originated in the awsmobile-cli source instead of this appsync starter app.

While you can test stuff offline, you will quickly run into the same problem while trying to publish `awsmobile publish`.

To "quick fix" the issue, 
1. Go to where the awsmobile-cli is installed (`where awsmobile`, open that awsmobile.cmd for a base path)
2. Install cross-spawn `npm i cross-spawn`
3. Replace all native NodeJS spawn() call with cross-spawn's (it's a drop-in replace)

**Comment by awsed** [comment]
I can confirm I don't get this error on mac or linux. Thanks for the help, @naginagi

**Comment by cloudDev12** [comment]
Hi @naginagi thanks for your suggestions above, however I'm not too sure where in the starter app code i should replace spawn() with cross-spawn (after doing the installation of cross-spawn). Sorry I'm very new to Angular/Node.js and just looking to get the demo starter app going so I can analyse each function to learn how its all done. Thanks

**Comment by cloudDev12** [comment]
Thanks for replying @naginagi. I replaced child_process with cross-spawn, however I still get the below error:

executing frontend start command ...
'ng' is not recognized as an internal or external command,
operable program or batch file.
frontend start command execution error:
{ Error: spawn ng ENOENT
    at notFoundError (C:\Users\jav_1\AppData\Roaming\npm\node_modules\awsmobile-cli\node_modules\cross-spawn\lib\enoent.js:6:26)
    at verifyENOENT (C:\Users\jav_1\AppData\Roaming\npm\node_modules\awsmobile-cli\node_modules\cross-spawn\lib\enoent.js:40:16)
    at ChildProcess.cp.emit (C:\Users\jav_1\AppData\Roaming\npm\node_modules\awsmobile-cli\node_modules\cross-spawn\lib\enoent.js:27:25)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:198:12)
  code: 'ENOENT',
  errno: 'ENOENT',
  syscall: 'spawn ng',
  path: 'ng',
  spawnargs: [ 'serve' ] }

PS C:\Users\jav_1\Documents\AWS Mobile apps built\aws-mobile-appsync-chat-starter-angular>

Is it because of the different locations? I'm really unsure on what to do sorry.

**Comment by naginagi** [comment]
@cloudDev12 do you have `@angular/cli` installed (either globally or locally as a devDependency)? Can you run `ng serve` in the appsync chat source code directory?

**Comment by naginagi** [comment]
@cloudDev12 by right you should be able to see all users in your Cognito User Pool. However, this is not relevance to the original issue in this thread. I suggest you create another ticket and let this issue be closed :)

**Comment by cloudDev12** [comment]
Ok will do. I will create another ticket for this issue and see how i go. thanks again for your help!!

**Comment by rojagit** [comment]
@naginagi Do we still have to go through this rigmarole for that node ENOENT error? Can you summarize with steps please ?
